### ISSUE TITLE
injection of TaskStorageConfig broken for tutorials

### ISSUE THREAD

**Issue by xvrl** [druid-io/druid#1212]
https://github.com/druid-io/druid/pull/1195 seems to have broken examples and tutorials.

```
2015-03-16T21:21:25,612 INFO [main] io.druid.initialization.Initialization - Loading extension[io.druid.extensions:druid-examples] for class[io.druid.initialization.DruidModule]
2015-03-16T21:21:25,614 INFO [main] io.druid.initialization.Initialization - Adding extension module[class io.druid.examples.ExamplesDruidModule] for class[io.druid.initialization.DruidModule]
2015-03-16T21:21:25,614 INFO [main] io.druid.initialization.Initialization - Loading extension[io.druid.extensions:mysql-metadata-storage] for class[io.druid.initialization.DruidModule]
2015-03-16T21:21:25,615 INFO [main] io.druid.initialization.Initialization - Adding extension module[class io.druid.metadata.storage.mysql.MySQLMetadataStorageModule] for class[io.druid.initialization.DruidModule]
Exception in thread "main" com.google.inject.CreationException: Guice creation errors:

1) Could not find a suitable constructor in io.druid.indexing.common.config.TaskStorageConfig. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at io.druid.indexing.common.config.TaskStorageConfig.class(TaskStorageConfig.java:29)
  while locating io.druid.indexing.common.config.TaskStorageConfig
    for parameter 0 at io.druid.indexing.overlord.HeapMemoryTaskStorage.<init>(HeapMemoryTaskStorage.java:60)
  at io.druid.cli.CliPeon$1.configureTaskActionClient(CliPeon.java:189)

1 error
	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:448)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:155)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
	at com.google.inject.Guice.createInjector(Guice.java:96)
	at com.google.inject.Guice.createInjector(Guice.java:73)
	at com.google.inject.Guice.createInjector(Guice.java:62)
	at io.druid.initialization.Initialization.makeInjectorWithModules(Initialization.java:369)
	at io.druid.cli.GuiceRunnable.makeInjector(GuiceRunnable.java:55)
	at io.druid.cli.CliPeon.run(CliPeon.java:205)
	at io.druid.cli.Main.main(Main.java:88)
```

reported in https://groups.google.com/forum/#!topic/druid-user/zxEwe-kiq-4

**Comment by xvrl** [comment]
@himanshug, would you mind having a look at this one? We should fix this before releasing 0.7.1

**Comment by himanshug** [comment]
@xvrl thanks for letting me know .I am looking
